### PR TITLE
mapping speed from grpc block update - reimplemented Lou's version

### PR DESCRIPTION
**Pull request by grooviegermanikus**
### Review Hints
- in the PR are three versions: the original (from_grpc_block_update_original), Dean's (from_grpc_block_update_optimized) and my rewrite (from_grpc_block_update_reimplement)
- the implementation is in from_grpc_block_update_reimplement+maptx_reimplemented
- tried to avoid multiple iteration over same vectors
- minimized cloning
- tried to make code more compact an readable - e.g. map_compute_budget_instructions
- added rayon (.into_par_iter()) which makes a huge difference but might have some inpact on tokio
- I have changed TransactionInfo to store the VersionedMessage struct instead of a base64-serialized version of it

### Benchmarks

(lite-rpc-staging->mainnet)

> from_grpc_block_update_reimplement{block.slot=251240678 num_transactions=1559}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=17.3ms time.idle=4.34µs
from_grpc_block_update_original{block.slot=251240678 num_transactions=1559}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=18.1ms time.idle=2.62µs
from_grpc_block_update_optimized{block.slot=251240678}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=15.2ms time.idle=2.43µs

> from_grpc_block_update_reimplement mapping took: 7.538083ms
from_grpc_block_update_original mapping took: 47.799542ms
from_grpc_block_update_optimized mapping took: 47.360167ms

> from_grpc_block_update_reimplement mapping took: 39.307583ms
from_grpc_block_update_original mapping took: 284.271834ms
from_grpc_block_update_optimized mapping took: 279.486084ms

> from_grpc_block_update_reimplement{block.slot=251230084 num_transactions=3232}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=33.1ms time.idle=4.60µs
from_grpc_block_update_original{block.slot=251230084 num_transactions=3232}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=42.2ms time.idle=3.11µs
from_grpc_block_update_optimized{block.slot=251230084}: solana_lite_rpc_cluster_endpoints::grpc_subscription: close time.busy=29.9ms time.idle=2.98µs

### CODE REVIEW

File: blockstore/src/block_stores/postgres/postgres_transaction.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl PostgresTransaction {
             prioritization_fees: value.prioritization_fees.map(|x| x as i64),
             cu_consumed: value.cu_consumed.map(|x| x as i64),
             recent_blockhash: value.recent_blockhash.clone(),
-            message: value.message.clone(),
+            message: BASE64.serialize(&value.message).expect("serialize message for database"),
```
Comment (godmodegalactus) [inline blockstore/src/block_stores/postgres/postgres_transaction.rs:41]: One idea could be avoid serialization if it isn not necassary.

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

**Comment by godmodegalactus** [comment]
Fix clippy

File: blockstore/src/block_stores/postgres/postgres_transaction.rs (hunk 1 above)
Comment (grooviegermanikus) [reply blockstore/src/block_stores/postgres/postgres_transaction.rs:41]: I moved that to this place. previously it was on the main mapper

**Comment by grooviegermanikus** [comment]
ported some improvements in PR https://github.com/blockworks-foundation/lite-rpc/pull/348

**Pull request closed without merging**